### PR TITLE
Feature/get group

### PR DESCRIPTION
**Pull request by diegolucasilva**

### CODE REVIEW

File: src/main/kotlin/com/maitre/menuservice/adapter/group/in/get/GetGroupHandlerImpl.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    override fun execute(serverRequest: ServerRequest): Mono<ServerResponse> {
+
+        logger.info(
+                "Get groups by menu_id Request initiated. " +
+                        "method=${serverRequest.method()}, path=${serverRequest.path()}, " +
+                        "headers=${serverRequest.headers()}, menu_id=${serverRequest.queryParam("menu_id")}")
+
+        val groups = getGroupsByMenuIdUseCase.execute(serverRequest.queryParam("menu_id")
+                .orElseThrow{RuntimeException("menu_id parameter is missing")}) //substituir por MissingParameterException
```
Comment (padovese) [inline src/main/kotlin/com/maitre/menuservice/adapter/group/in/get/GetGroupHandlerImpl.kt]: Can be done already.

File: src/main/kotlin/com/maitre/menuservice/adapter/group/in/GroupRouter.kt [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
 class GroupRouter(
         private val createGroupHandler: CreateGroupHandler,
+        private val getGroupHandler: GetGroupHandler,
 ) {
 
     @Bean
     fun routerGroup() = router {
         POST("/group", createGroupHandler::execute)
+        GET("/group", getGroupHandler::execute)
     }
```
Comment (padovese) [inline src/main/kotlin/com/maitre/menuservice/adapter/group/in/GroupRouter.kt]: For convention, let's equalize the names:

Class MenuRouter, method menuRoutes
Class ProductRouter, method productRoutes
Class **GroupRouter**, method **groupRoutes**

Shall we?

File: src/main/kotlin/com/maitre/menuservice/domain/group/usecases/GetGroupsByMenuIdUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+@Service
+class GetGroupsByMenuIdUseCase(private val logger: Logger,
+                               private val getGroupsByMenuIdPort: GetGroupsByMenuIdPort) {
+
+    fun execute(menuId: String): Flux<Group> {
+        logger.info("Get groups by menu_id use case initiated. id=$menuId")
+        return getGroupsByMenuIdPort.getByMenuId(menuId)
+                .switchIfEmpty(Mono.error(GroupNotFoundException(menuId)))
+                .doOnNext { logger.info("id=${menuId}, group=${it.toJson()}") }
```
Comment (padovese) [inline src/main/kotlin/com/maitre/menuservice/domain/group/usecases/GetGroupsByMenuIdUseCase.kt:22]: Aren't you writing the "Verify if menu exists" validations?
We can open a issue.

**Review by padovese — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/maitre/menuservice/adapter/group/in/GroupRouter.kt (hunk 2 above)
Comment (diegolucasilva) [reply src/main/kotlin/com/maitre/menuservice/adapter/group/in/GroupRouter.kt]: done

File: src/main/kotlin/com/maitre/menuservice/adapter/group/in/get/GetGroupHandlerImpl.kt (hunk 1 above)
Comment (diegolucasilva) [reply src/main/kotlin/com/maitre/menuservice/adapter/group/in/get/GetGroupHandlerImpl.kt]: done

**Review by padovese — APPROVED** [review]

**Pull request merged**